shared/utils/dateFilterUtils.ts

```diff
@@ -0,0 +1,41 @@
+import { DateTime } from 'luxon'
+
+import { EventModel } from '../api'
+import { MAX_DATE_RECURRENCES } from '../constants'
+
+export type UseDateFilterReturn = {
+  startDate: DateTime | null
+  setStartDate: (startDate: DateTime | null) => void
+  endDate: DateTime | null
+  setEndDate: (endDate: DateTime | null) => void
+  filteredEvents: EventModel[] | null
+  startDateError: string | null
+}
```

Comment:
:wrench: This belongs in `hooks/useDateFilter`

Reply:
Sure ...I moved it here just for the codeClimate complaning about the number of lines in the hook.

Reply:
Are you sure this is about the type? I think code climate only complains about functions that are too long, not about files that are too long. We have a few files with 200+ lines of code, so this should not matter. Extracting the functionality like you did below is totally fine, just this type does not belong here imo.